### DISCUSSION TITLE
Portfolio Performance Contest

### DISCUSSION THREAD

**Discussion by bear1bear2bear3** [asastats/channel#61] [General]
### Hi guys, we have been working on a new form of contest for the community. Here is the proposal:

_please let us know what you think about it, what could be done better, and how you can help_

------------------------------------
Basically contestants will be able to create **mock portfolios** at the start of the contest and the best performing portfolio after a period of 2 (?) weeks wins!

The best 3(?) will receive the proceedings from all created portfolios as a **prize**!!

To participate, all that needs to be done is to post your contest portfolio in the comments of the contest post on reddit. Every contestant will get the chance to invest 1 ALGO in their portfolio (Dont worry, funds will be covered by ASAStats, so you dont have to spend your own ALGOs). A max of 7 assets can be chosen. The format below should be followed so that our (yet to be created) script can read your comment and create a portfolio from it.

Format:
10% Asset A
20% Asset B
50% Asset C
10% Asset D
10% Asset E
_This is just a preliminary format. It is up for discussion and can be adjusted to the dev’s/script’s needs_

The way this could work is that a script picks up all contest portfolios by screening the comments of a reddit post. It then creates the portfolios in real, new Algorand addresses in the correct ratio with a total budget of 1 ALGO per portfolio. This will be paid from the admin pool (or whatever pool is suited best?). The transaction cost as well as network fees shall be covered by that pool as well, in order to have the correct amount of funds in the individual addresses/portfolios.

Once all portfolios are created, a list of all contestants and their portfolios will be published. There will be regular updates on the ranking. These will include the wallet addresses as well as a link to the corresponding ASAStats page in order for users to easily check the value, assets and so on. 

After 2 weeks, all assets will be **sold**, the initial investment goes back to the admin pool and the **profits** (that will have hopefully happened) will be **distributed** to the top 3 contestants.

_Ipalekas comments on the script and format of publishing were as follows:_

> So let's say every participant gets 1 Algo and then they comment on their portfolios like:
> 10% ALGO
> 20% ASASTATS
> 50% YLDY
> 10% goBTC
> 10% Ghost
> And we create and then update our main file like:
> {
>   "u/somebody": {
>     "393537671": 200000,
>     "226701642": 500000,
>     "386192725": 100000,
>     "714773890": 100000,
>   }
> }
> So at the beginning of the contest, the script will traverse all the entries and locate all the tokens and sum up related microAlgos for specific tokens, then the script will buy those tokens spending exactly the amount of microAlgos collected. In the end, it will allocate tokens to the users based on percentage of the user's microAlgos of the sum of microAlgos spent for purchasing those tokens.
> We will create posts in our community channels and will sort them once a day or so, like:
> ... 5. u/somebody 1.24 Algo ...
> I'm talking about the script as we can repeat this contest over and over, on a monthly basis or so.

--------------------------------

_You can find the first short conversation about this contest here: https://www.reddit.com/r/asastats/comments/ubrf16/trading_contest_take_2/_

--------------------------------
### What do we need?
-Dev who writes the script
-Announcement post and proof reading (especially to get a feeling about the communities interest)
-?

**Comment by ipaleka** [comment]
I propose a monthly cycle and adding up a broader contest for governance spots sponsored by the admpool.

**Reply by bear1bear2bear3** [reply]
Monthly works too! But i think we need a "cool off" period or smth. Where everything gets sorted, prizes get paid out, contestants can take a few days to observe the market or smth, idk
Maybe 3 weeks where performance is measured and 1 week for all the stuff around it

**Reply by ipaleka** [reply]
That's a valid point. I guess then a week for preparation and another week to finish the previous cycle would define a month cycle with two weeks for the actual price movement.

**Comment by dragmz** [comment]
I suggest to register participants first and then automate (aka write scripts if any needed).

Can participants decide when to exchange their ALGO for ASA? Timing the market correctly is important.

**Reply by ipaleka** [reply]
The simplest solution for the contest is that scripts swap *at the same time* all the tokens. They only can have an option to PR their entries and change microAlgos spent for some tokens.

**Reply by bear1bear2bear3** [reply]
So you think it would be better to have everyone sign up for it first and then create the portfolios in a second step?

Obviously i dont know how difficult it would be, but i think it would be amazing if people could participate with a single comment where they just say their preferred ratios. I feel like this would increase participation by magnitudes bc it is as simple and "no strings attached" as it gets

**Reply by ipaleka** [reply]
It would be the simplest way, so implies your suggestion is that. I'm just saying that they should be able to change their values (at least) on GitHub once they give the initial value. From the game perspective that doesn't change anything about outcomes.

So they can change on 15., 21., but then the results are frozen on 28. and those values will be spent to buy tokens on 28. at 18:00.

**Reply by bear1bear2bear3** [reply]
Ahhhh so you want them to be able to trade at least a little bit? Also a nice idea!

**Reply by ipaleka** [reply]
Not really trading, just a way to change their minds before official starts. This 28. represents the official start date.

The main factor here is the number of participants and we should allow them a flowless contest. I guess we can add prizes in ASASTATS too.

**Reply by bear1bear2bear3** [reply]
Ah now i understand. They should be able to change during the period of signing up (=presenting their portfolio for the first time) and the actual start of the contest. That makes sense, bc otherwise people might wait until the very last minute and we wont be able to estiment the # of contestants until then

**Reply by bear1bear2bear3** [reply]
28.5. should be the start pf the contest?

**Reply by ipaleka** [reply]
> 28.5. should be the start pf the contest?

We first should ensure there are enough human resources to conduct everything. We need a few people to check entries a few times a day, to coordinate current listings, communicate with DragMZ, etc.

Also, would we go with a Gitcoin bounty or a Reddit post as a starting point?

P.S. we also haven't discussed this subthread's initial suggestion by DragMZ, do we need to register people who are willing to participate in the contest? If we can't get enough "subscribers" in a reasonable amount of time, then we should probably give up on this.

**Reply by bear1bear2bear3** [reply]
Mhm yes youre right, the effort put into this to make it work well will be high and if it only reaches a few people its not worth it. 

Creating the portfolios is one thing but keeping the community updated on the ranks etc is another. Would it be possible to somehow integrate the ranking into the website? So one could pull the info from there rather than having to compare all wallets? The thing is that if we want to do this for months we probably wont find anybody that checks entries and posts the current ranking every day. Only works if most of it automated and all that needs to be done is post a link to the ranking every couple days. Thats just from my POV tho.

Without having any real clue .. it feels like this would be a lot of upfront coding for a not so sure success. I love the idea of the contest but obviously we cant know if it gets traction.
IMO the one downside of a post thats asks people if they are interested is that we are missing out on the USP of the concept, namly that a user sees the post, thinks it sounds fun and simply creates a comment with their portfolio. I feel like this kind of stuff needs to have the lowest barrier of entry possible or else we'll end up like the first contest. On the other hand, yes, that means effort prior to knowing whether it'll be usefull.

**Reply by bear1bear2bear3** [reply]
I know you are asking whether someone has time to do the administrative stuff, i could check in more often. However i couldnt check more often every day for months. I dont want to sound like and idiot and put all the work on you either. Its just that realistically i think the work needed will not be done if the contest has to rely on people working for it every day

**Reply by ipaleka** [reply]
Thank you for your straightforward thoughts, that's exactly what we need here. :)

Tbh, I won't be doing a lot with this, my involvement would probably come down to discussion like this one. And that implies no website page for this.

And for the subscribing issue, if people want they can in their initial comment give their formal percentages too - they can change them if the starting date is prolonged too much.

Afaic if we want for this to succeed then we have to organize people as you did for Decipher event. We'd arrange a rewards budget for that. The only condition I set is that all of your discussion should take place in public space, preferably here under a dedicated discussion thread. I guess I can create a "Current standings" script that will list participants sorted, so we can use that.

**Reply by bear1bear2bear3** [reply]
True a task force kinda thing would be necessary and is a good idea. 

Ofc everything will be discussed publicly now.

How and where should we start looking for poeple? I can imagine that some of the poeple who did decipher would be interested again

**Reply by ipaleka** [reply]
> How and where should we start looking for people?

By making suggestions and discussing them. :)

**Reply by bear1bear2bear3** [reply]
ad subscribing: i think your proposal should be sufficient. People can already "register" their portfolio in the first announcement post but also in the next post that would mark the actual start. Sounds good.

tbh i think it would be easiest do ask on discord. Reddit and twitter will likely require another onboarding process whereas most people who are still (somewhat) active on our discord already know how to operate. 

I guess the amount of people needed should be considered too. Therefore we kind of need a list of tasks that are expected to emerge and need to be done by helpers.

- create contest wallet and manage funds
- create announcements
- screen announcements
- create portfolios & capture initial values (for documentation and transparency)
- link portfolios to usernames/handles and keep this document organised
- regularly check all portfolios and rank them (except if you make the script, if i understand correctly)
- regularly post current ranking
- create announcement for changing portfolios (once?) half way through
- screen comments on that post & adjust portfolios
- capture final values & sell everything
- name a winner and send out rewards
- ongoing community management 

feel free to add / scratch points.

i think the most time intensive will be the creation of wallets (writing down seed phrases, keeping everything organised so we don't lose access,..) and the purchases (especially if we bundle purchases and divide them up later). Those are also tasks that might not be done by every helper because it requires access to funds? On the other hand it wont be much in ALGO or EURO terms I guess, so maybe its not that critical. 

looking at the tasks i'd say a team of around 7-10 is needed, depending obviously on the number of participants. And most helpers will be needed for creating the wallets and portfolios and then selling everything again, so at the start and end of the whole process.

**Reply by ipaleka** [reply]
Just a note that we'd do **everything** related to accounts/addresses by a script or we won't have this contest, there's no third option.

I guess the current standings table would also be created by a script, so besides running the scripts only creating posts from them will be manual work.

**Reply by bear1bear2bear3** [reply]
Now you are talking ipaleka!
Thats exactly what i was hoping for. 🚀

Did i maybe just understand you incorrectly earlier?

Then i guess we should be fine with 2-3 people handling the announcements. I will take part in it and will find myself some helpers

**Comment by ipaleka** [comment]
by [DragMZ](https://github.com/asastats/channel/discussions/61#discussioncomment-2708278):  
> I suggest to register participants first and then automate (aka write scripts if any needed).

So this discussion is actually a discussion about our approach to this.

We can either do what @dragmz suggests, start with a post in our subreddit, and eventually with r/Algorand and/or r/AlgorandOfficial posts. If there would be enough contesters - dozens or hundreds rather than a few - we can go with scripting the whole thing.

The other approach is that we craft a Gitcoin bounty article and program and we advertise this contest there.

In both cases afaic we have to define additional ASASTATS rewards for the winners.

But, first of all, we have to discuss and analyze our resources in this. Besides crafting an eventual Gitcoin post and script development, there will be administration work involved too. Do we have such human resources?

**Reply by Sammyspeed75** [reply]
Guys, Any update about this task?

**Reply by ipaleka** [reply]
by bear1bear2bear3 in [Discord](https://discord.com/channels/906917846754418770/911091925174607935/978358021451288576):  
> We need a few people to help with the Portfolio Performance Contest. The tasks would mainly be writing and posting announcements and doing some community management revolving around the contest. If someone is keen to help, please read the discussion and comment there that you want to help. I guess we‘ll create an issue after seeing if / how many people are joining.
> 
> https://github.com/asastats/channel/discussions/61

**Reply by Sammyspeed75** [reply]
Alright great, I'm keen to help out with the contest.

**Comment by ipaleka** [comment]
@bear1bear2bear3 isn't around, but competitions are organized from outside. :)

https://twitter.com/TigerChiClub/status/1567980733914042368
> On the first day of our trading competition, our tiger traders shared a good site for checking the overall balance of an account. 
Check  [@asastatscom](https://twitter.com/asastatscom) out! 
Each of our trader was given a wallet of 5 tokens worth 580ALGO! Let's see how they do in a few days time!
$ALGO

![FcKWFbtWAAIQtR-](https://user-images.githubusercontent.com/49662536/189373057-b596a2fa-7749-4d05-a7c2-25b6165cfb4a.jpeg)
